test/unit_tests/braket/circuits/test_circuit.py

```diff
@@ -570,6 +572,193 @@ def test_add_verbatim_box_result_types():
+        .add_instruction(Instruction(compiler_directives.EndVerbatimBox()))
+        .add_instruction(Instruction(Measure(), 0))
+    )
+    expected_ir = OpenQasmProgram(
+        source="\n".join(
+            [
+                "OPENQASM 3.0;",
+                "bit[1] b;",
+                "qubit[2] q;",
+                "#pragma braket verbatim",
+                "box{",
+                "x q[0];",
+                "x q[1];",
+                "}",
+                "b[0] = measure q[0];",
+            ]
+        ),
+        inputs={},
+    )
+    assert circ == expected
+    assert circ.to_ir("OPENQASM") == expected_ir
+
+
+def test_measure_in_verbatim_subcircuit():
+    message = "Cannot measure a subcircuit inside a verbatim box."
+    with pytest.raises(ValueError, match=message):
+        Circuit().add_verbatim_box(Circuit().x(0).x(1).measure(0))
+
+
+def test_measure_empty_circuit():
+    with pytest.raises(IndexError):
+        Circuit().measure()
+
+
+def test_measure_no_target():
+    circ = Circuit().h(0).cnot(0, 1).measure()
+    expected = (
+        Circuit()
+        .add_instruction(Instruction(Gate.H(), 0))
+        .add_instruction(Instruction(Gate.CNot(), [0, 1]))
+        .add_instruction(Instruction(Measure(), 0))
+        .add_instruction(Instruction(Measure(), 1))
+    )
+    assert circ == expected
+
+
+def test_measure_with_result_types():
+    message = "a circuit cannot contain both measure instructions and result types."
+    with pytest.raises(ValueError, match=message):
+        Circuit().h(0).sample(observable=Observable.Z(), target=0).measure(0)
+
+
+def test_result_type_with_measure():
+    message = "Cannot add a result type to a circuit which already contains a measure instruction."
+    with pytest.raises(ValueError, match=message):
+        Circuit().h(0).measure(0).sample(observable=Observable.Z(), target=0)
+
+
+def test_measure_with_multiple_measures():
+    circ = Circuit().h(0).cnot(0, 1).h(2).measure([0, 1]).measure(2)
+    expected = (
+        Circuit()
+        .add_instruction(Instruction(Gate.H(), 0))
+        .add_instruction(Instruction(Gate.CNot(), [0, 1]))
+        .add_instruction(Instruction(Gate.H(), 2))
+        .add_instruction(Instruction(Measure(), 0))
+        .add_instruction(Instruction(Measure(), 1))
+        .add_instruction(Instruction(Measure(), 2))
+    )
+    assert circ == expected
+
+
+def test_measure_same_qubit_twice():
+    message = "cannot measure the same qubit\\(s\\) 0 more than once."
+    with pytest.raises(ValueError, match=message):
+        Circuit().h(0).cnot(0, 1).measure(0).measure(1).measure(0)
+
+
+def test_measure_empty_measure_after_measure_with_targets():
+    message = "cannot perform multiple measurements of the same qubits."
+    with pytest.raises(ValueError, match=message):
+        Circuit().h(0).measure(0).measure()
+
+
+def test_measure_gate_after():
+    message = "Cannot add a gate or noise after a measure instruction."
```

Comment:
It sounds like we could reduce this constraint to only prevent ops on measured qubits. I think that would be valuable if you feel like the implementation won't be too complicated

Reply:
Added an error when the `target`, `target_mapping` or `instruction.target` overlap with `self._measured_targets`